src/main/java/org/apache/maven/plugins/assembly/AssemblerConfigurationSource.java

```diff
@@ -223,4 +223,9 @@ public interface AssemblerConfigurationSource {
      * @return Override group name.
      */
     String getOverrideGroupName();
+
+    /**
+     * @return Override umask.
+     */
+    Integer getOverrideUmask();
```

Comment:
Adding a method to an interface is an incompatible change in public API that can break existing code. In Java 8 this should have a default implementation to avoid that.

Reply:
Sorry, but I failed to understand why do we need to support backward compatibility for this interface. What about `org.apache.maven.plugins.assembly.AssemblerConfigurationSource#getOverrideGroupName` method (which I followed in this pull request)?

I was under impression that we need to provide backward compatibility for plugin configuration from POM point of view only - this is the reason `null` value is supported (and means that respective configuration options is not defined).

Reply:
You never know who your dependents are. It's easy to add a return null here, and then you don't break anyone.

Reply:
Added default method implementation.